cmd/gc.go

```diff
@@ -217,7 +217,7 @@ func gc(ctx *cli.Context) error {
 		if obj.IsDir() {
 			continue
 		}
-		if obj.Mtime().After(maxMtime) || obj.Mtime().Unix() == 0 {
+		if obj.Mtime().UnixNano() > maxMtime.UnixNano() || obj.Mtime().Unix() == 0 {
```

Comment:
Why this?

Time.After() is implemented using Unix timestamp
```
func (t Time) After(u Time) bool {
	if t.wall&u.wall&hasMonotonic != 0 {
		return t.ext > u.ext
	}
	ts := t.sec()
	us := u.sec()
	return ts > us || ts == us && t.nsec() > u.nsec()
}
```

The bug may came from how we create the obj.Time() with wrong timezone.

Reply:
when the backend storage is minio, The obj.Mtime() may return the UTC timezone, and time.Now() return current timezone

Reply:
That does not matter, After() works even t and u in different timezone.

And we use Time.After() in unit test [1], and MinIO pass the unit tests, so I think this is not the problem.

[1] https://github.com/juicedata/juicefs/blob/main/pkg/object/object_storage_test.go#L109